chia/farmer/farmer_api.py

```diff
@@ -73,41 +77,170 @@ async def new_proof_of_space(
                 sp.difficulty,
                 new_proof_of_space.sp_hash,
             )
-            # Double check that the iters are good
-            assert required_iters < calculate_sp_interval_iters(self.farmer.constants, sp.sub_slot_iters)
 
-            # Proceed at getting the signatures for this PoSpace
-            request = harvester_protocol.RequestSignatures(
-                new_proof_of_space.plot_identifier,
-                new_proof_of_space.challenge_hash,
-                new_proof_of_space.sp_hash,
-                [sp.challenge_chain_sp, sp.reward_chain_sp],
-            )
+            # If the iters are good enough to make a block, proceed with the block making flow
+            if required_iters < calculate_sp_interval_iters(self.farmer.constants, sp.sub_slot_iters):
+                # Proceed at getting the signatures for this PoSpace
+                request = harvester_protocol.RequestSignatures(
+                    new_proof_of_space.plot_identifier,
+                    new_proof_of_space.challenge_hash,
+                    new_proof_of_space.sp_hash,
+                    [sp.challenge_chain_sp, sp.reward_chain_sp],
+                )
 
-            if new_proof_of_space.sp_hash not in self.farmer.proofs_of_space:
-                self.farmer.proofs_of_space[new_proof_of_space.sp_hash] = [
-                    (
-                        new_proof_of_space.plot_identifier,
-                        new_proof_of_space.proof,
+                if new_proof_of_space.sp_hash not in self.farmer.proofs_of_space:
+                    self.farmer.proofs_of_space[new_proof_of_space.sp_hash] = [
+                        (
+                            new_proof_of_space.plot_identifier,
+                            new_proof_of_space.proof,
+                        )
+                    ]
+                else:
```

Comment:
Appears you can just set the value to emptylist in the if-block and append unconditionally.